### PR TITLE
Fix: validate preserved active tab against currently visible tabs

### PR DESCRIPTION
**Pull request by jcPimcore**
## Summary
Follow-up to #3997, addressing review feedback raised on the [`2026.2` backport PR #3998](https://github.com/pimcore/studio-ui-bundle/pull/3998#discussion_r3815192323).

- #3997 made `refreshElement` preserve the previously active tab across a data-object refresh. That preservation is unconditional: it can restore a tab key that becomes hidden by the very same refresh — e.g. the "workflow" tab, which is hidden via `hidden: () => !isWorkflowAvailable(element, elementType)` in `tabs-container.tsx`, and `isWorkflowAvailable` depends on `element.hasWorkflowAvailable`, a field the API can flip after a refetch (e.g. right after a workflow transition, via `EditorWorkflowSubjectBridge.onApplied` → `refreshElement`).
- `EditorTabs`'s only self-correction path only fires when `activeTab === null`. A stale-but-non-null preserved key that's no longer in the visible `items` was passed straight through to antd's `Tabs` as `activeKey`, leaving the editor with **no tab visibly selected** — worse than the original bug.
- Fix: the tab-reset effect in `editor-tabs.tsx` now also fires when the current `activeTab` is no longer present among the visible tabs, not only when it's `null`.

### Scope note

`editor-tabs.tsx` hosts the **shared** `EditorTabs` component, used by the Asset and Document editors as well as Data Object — so this fix changes tab-fallback behavior for all three element types, not only Data Object (which is where #3997's original bug report was scoped). This is intentional and safe: the component doesn't branch on element type anywhere in this logic, the fix only activates when a previously-active tab is no longer visible (a real edge case for any element type with conditionally-hidden tabs), and it's covered by `editor-tabs.test.tsx`, which exercises the shared logic directly rather than a data-object-specific path.

A second, unrelated commit (`test: narrow mock element type from any to Partial<IElementDraft>`) is pure test cleanup addressing separate review feedback, no behavior change.

Mirrored identically on the `2026.2` backport PR #3998.

## Test plan
- [x] Added `editor-tabs.test.tsx` covering: defaults to first tab when none active, keeps a still-visible active tab, falls back to the first visible tab when the preserved tab is no longer visible, keeps a preserved tab that stays visible.
- [x] Verified the new test fails without the fix and passes with it (reverted the one-line fix locally, confirmed the "falls back..." test fails; restored and re-confirmed green).
- [x] `npx tsc --noEmit`, eslint, and the full jest suite (35 suites / 346 tests) all passed on this exact diff, verified on a sibling branch immediately before this one was cut (identical single-hunk diff, same test file). The follow-up type-narrowing commit was verified with eslint (type-aware) and the scoped `editor-tabs.test.tsx` suite (4/4) — a fresh full-project `tsc`/full-suite rerun on this exact branch was blocked by transient Docker/host memory exhaustion from unrelated containers on the machine, not by anything in this change.
- [ ] Manual verification in a running Studio UI instance (not performed in this environment).

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: assets/js/src/core/modules/element/editor/tabs/editor-tabs.test.tsx [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+ * Pimcore Open Core License (POCL)
+ * Full copyright and license information is available in
+ * LICENSE.md which is distributed with this source code.
+ *
+ *  @copyright  Copyright (c) Pimcore GmbH (https://www.pimcore.com)
+ *  @license    Pimcore Open Core License (POCL)
+ */
+
+const mockSetActiveTab = jest.fn()
+const mockElementDraftState: { activeTab: string | null, element: any } = {
```
[the remark this answers is not in the document]
Comment (jcPimcore) [reply assets/js/src/core/modules/element/editor/tabs/editor-tabs.test.tsx]: Good catch — fixed in 46e012c91, narrowed the mock element type (line 12) and the WORKFLOW_TAB hidden callback param (line 79) from `any` to `Partial<IElementDraft>`. Applied the same fix to #3998. Verified with eslint and the test suite; `tsc --noEmit` on the full project is currently blocked by unrelated host memory exhaustion in this environment (other containers), so I couldn't get a fresh full-project rerun, but eslint's type-aware check on this file is clean.
